Update README with contribution guidelines and defined directory structure

rancher-max opened the pull request.

rancher-max commented on ansible/rancher/default-ha/rancher-playbook.yml:
```diff
@@ -8,7 +8,7 @@
     terraform_workspace: "{{ lookup('env', 'TF_WORKSPACE') }}"
     # Construct the path to the workspace's state file
-    terraform_state_file: "terraform/aws/cluster_nodes/terraform.tfstate.d/{{ terraform_workspace }}/terraform.tfstate"
+    terraform_state_file: "terraform/aws/modules/terraform.tfstate.d/{{ terraform_workspace }}/terraform.tfstate"
```
yeah I agree with that! I'll leave this unresolved until I make the changes and let others weigh in if they want to.

izaac commented on README.md:
````diff
@@ -1,14 +1,30 @@
+See below for a visual representation of the directory structure. In the ansible directory, a product might be "rke2" or "rancher", and a feature might be "airgap." Scripts are there for reusable scripts, for example to run the set of commands required to install an rke2 server node. In the terraform directory, a provider might be "aws" or "harvester", and modules-with-context might be "airgap-mo ...
 
 ```
 ├── ansible/
````
cc @ShylajaDevadiga

rancher-max replied:
I'd argue that there shouldn't need to be "team-specific" things ever in here -- everything should be usable by any team, so should follow the structure to make it clear what specifics it might be doing. For example, if a team wanted to bring up a rancher instance with some custom setting pre-enabled, maybe that's just an optional variable to pass in the configuration variables, but just using a standard `ansible -> rancher -> default-ha` playbook.

izaac replied:
I am getting a bit more familiar, it makes sense though I am still a bit concerned of ending with circular conflicts (deps?) in the future. I'll think more about this once I get a bit more familiar with the structure.

igomez06 replied:
I agree with Max, I think just having descriptive product names should be sufficient

rancher-max commented on README.md:
````diff
@@ -1,14 +1,30 @@
 ```
 ├── ansible/
-│   └── rke2/
+│   ├── product/
````
Yep! Whatever product the ansible playbook is creating!

rancher-max commented on README.md:
```diff
@@ -1,14 +1,30 @@
-## Current Directory Structure
+## Contributing
+
+All are welcome and encouraged to contribute! Try to keep any changes generalized, easy to understand, and reusable by all. There may be cases, however, where some specifics are required. In these cases, make sure to include yourself or your team in the CODEOWNERS file for the necessary path. Any new ansible playbook or terraform module should have a README that explains usage, including input pa ...
```
This link is the right location for understanding more about it. Do you want me to include this link? There isn't a CODEOWNERS file yet in the repo itself.

izaac replied:
IMO I'd rather have it if the CODEOWNERS is already mentioned in the doc

rancher-max replied:
Added one in! It's not perfect since I'm not sure that PIT should actually own that existing file, but it's a good reference.

igomez06 reviewed (approved):
LGTM

Felipalds reviewed (commented):
Hey team, based on some conversations I had with @rancher-max and mainly with @igomez06 when I first tried the repository, I'll drop some things here. It was some time ago so maybe not all be necessary:

1. It would be good to document that we need to update the `/etc/hosts` from our machine in order to make it accessible from the host;
2. Also good to document that we need to create the Security Group, subnets and VPC manually on AWS (a terraform file for this would be nice);
3. You will face an error if try to run Rancher Playbook with 2 workers and 1 master (if the master node has no worker role);
4. Sometimes you will need to export the `TF_WORKSPACE` to the workspace you are working on. Error is: `(Could not find state_file \"/home/luiz-rosa/suse/qa-infra-automation/terraform/aws/cluster_nodes/terraform.tfstate.d/terraform.tfstate\`;
5. Good to know that we from SSE are on an AWS account where our hosted zone will be the `qa-test.rancher.space` and not that default. Just point that on documentation to avoid questions;
6. And also good to point that the best way to recreate the environment is destroy all with Terraform and recreate all again;
7. I faced a dumb issue on my side because I created a machine with _ubuntu_ user and forgot to change on the variables xD, then I couldn't run anything because the SSH failed. Maybe add a "Debug" area on the README;
8. Last one should document that the user_id should be a string that you define to identify yourself or the specific deployment context. It's not a system-generated ID or an AWS-specific user ID in this context. Pointing this because I had questions too.

I know some of these are simple/dumb points but I think adding those on the documentation should avoid "time-losing" with questions and debugging. Also, as the repo will be wide used on organization, more people from SSE, QA and Support may be using it as well :) 

WDYT? Thanks!!

![image](https://github.com/user-attachments/assets/43a3105e-f5cf-4848-8432-5d886339e789)

rancher-max commented:
Hey @Felipalds thanks for the review! Some responses below:


> 1. It would be good to document that we need to update the /etc/hosts from our machine in order to make it accessible from the host;

Can you elaborate on this one for me? I don't think I have to actually do this 👀 

> 2. Also good to document that we need to create the Security Group, subnets and VPC manually on AWS (a terraform file for this would be nice);

This is a good call and an easy change, so I made the change to the associated module's README.

> 3. You will face an error if try to run Rancher Playbook with 2 workers and 1 master (if the master node has no worker role);

I'm going to leave this change for someone else. There may be updates that need to made to the module itself, so I don't feel it's good for me to just update the README on that playbook, and this PR is really just focusing on documentation.

> 4. Sometimes you will need to export the TF_WORKSPACE to the workspace you are working on. Error is: (Could not find state_file \"/home/luiz-rosa/suse/qa-infra-automation/terraform/aws/cluster_nodes/terraform.tfstate.d/terraform.tfstate\;

I think this might have been covered by a different PR! I've updated the relevant language in the README to account for having a "file with default environment variables." Let me know if that resolves this!

> 5. Good to know that we from SSE are on an AWS account where our hosted zone will be the qa-test.rancher.space and not that default. Just point that on documentation to avoid questions;

This is a great point in general. People using this will often have different hosted zones, so we should try to make these variables where possible. I updated the "Standards" in the README for this.

> 6. And also good to point that the best way to recreate the environment is destroy all with Terraform and recreate all again;

This is may be helpful to add to the FAQ. I'm leaving it out for now but please feel free to submit a PR after this goes in with the addition!

> 7. I faced a dumb issue on my side because I created a machine with ubuntu user and forgot to change on the variables xD, then I couldn't run anything because the SSH failed. Maybe add a "Debug" area on the README;

Ooo I think a whole debug doc could be helpful. I'm again going to leave it out for now, but PRs welcome 😄 

> 8.     Last one should document that the user_id should be a string that you define to identify yourself or the specific deployment context. It's not a system-generated ID or an AWS-specific user ID in this context. Pointing this because I had questions too.

I think the change I will make for this is to add a standard to add descriptions to the variables.


Take a look -- I've pushed all the changes up!

slickwarren commented on CODEOWNERS:
```diff
@@ -0,0 +1 @@
+/tofu/aws/modules/cluster_nodes @rancher/qa-pit-crew
```
we need to make sure any groups added here are part of / have permission to this repo

rancher-max replied:
Yep! This team has write access to this repo now 🎉 

As an aside, we will need to add other teams as well when any are added to the CODEOWNERS file here

slickwarren commented:
otherwise lgtm

Felipalds reviewed (commented):
LGTM! I will create a PR with Q/A and "debug" section and you take a look to see if it is worth to add to the repo @rancher-max. Thank you for your response xD
